src/main/java/com/devonfw/ide/sonarqube/common/impl/check/component/DevonArchitectureComponentDeclarationCheck.java

```diff
@@ -15,9 +14,10 @@
  * {@link DevonArchitectureComponentCheck} verifying that a {@link Component} itself is properly defined in
  * {@link Architecture} and that the general {@link Component#getDependencies() dependencies} are not violated.
  */
-@Rule(key = "Devon4j:C1", name = "Devon Component Declaration Check", //
-    description = "Verify component is properly declared in architecture.json file.", //
-    priority = Priority.CRITICAL, tags = { "architecture-violation", "devonfw", "component" })
+// @Rule(key = "Devon4j:C1", name = "Devon Component Declaration Check", //
+// description = "Verify component is properly declared in architecture.json file.", //
+// priority = Priority.CRITICAL, tags = { "architecture-violation", "devonfw", "component" })
+@Rule(key = "C1")
```

Comment:
Why do we have to change the rule keys? We have references to them in external files that you did not change and that will break with the PR.
Also the key of a rule has IMHO to be unique all across SonarQube. How can we assure that without a namespace like `Devon4j:`?
Existing rules from SonarQube are called `Squid:«number»`. I considered the rules keys from the makers of SonarQube as a kind of best-practice to follow.
If you have good reasons to change the keys please explain here. Otherwise I would suggest to keep the old keys.

Reply:
The problem with the external files is a good point, haven't thought of that. As for the uniqueness of the rules: Our rules are saved under our repository key `devon-java`, so e.g. this rule will be shown in SonarQube as `devon-java:C1`. The SonarQube rules probably belong to the repository `Squid`. Would it suffice for the external files to maybe change our repository key to `Devon4j`? Then our rules would still have the same name as they had before.